gui/widgets/editable.cpp

```diff
@@ -63,15 +63,21 @@ void EditableWidget::reflowLayout() {
 }
 
 void EditableWidget::setEditString(const String &str) {
-	// TODO: We probably should filter the input string here,
-	// e.g. using tryInsertChar.
-	_editString = str;
+	_editString = "";
+	int pos = 0;
+	for (unsigned int i = 0; i < str.size(); i++)
+		if (tryInsertChar((byte)str[i], pos))
+			pos++;
 	_caretPos = 0;
 }
 
 bool EditableWidget::tryInsertChar(byte c, int pos) {
 	if ((c >= 32 && c <= 127) || c >= 160) {
-		_editString.insertChar(c, pos);
+		if (_maxLength < 0 || _editString.size() < (unsigned int)_maxLength)
+			_editString.insertChar(c, pos);
+		else
+			_caretPos--;
```

Comment:
The calling code increments the position of _caretPos on it's own, so this is to counter-act that. If I don't decrement it here, it will cause the caret to be incremented even when no character is typed.